lib/server.js

```diff
@@ -81,6 +81,29 @@ exports.create = function create(createOptions) {
       callback(err, html);
     }
 
+    function matching(options) {
+      var componentInstance = null;
+      match(options, function(error, redirectLocation, renderProps) {
+        if (redirectLocation) {
+          debug('server.js match 302 %s', JSON.stringify(redirectLocation));
```

Comment:
@samsel i think it's there for any redirection framework, it might go in an infinite loop. i have test it, but we can easily catch it if we match the options.location with the redirectionLocation to throw an error.